enterprise_access/apps/events/utils.py

```diff
@@ -0,0 +1,147 @@
+temporary.
+"""
+
+import logging
+
+from confluent_kafka import KafkaError, KafkaException, SerializingProducer
+from confluent_kafka.admin import AdminClient, NewTopic
+from confluent_kafka.error import ValueSerializationError
+from confluent_kafka.serialization import StringSerializer
+from django.conf import settings
+
+from enterprise_access.apps.events.data import CouponCodeRequestEvent, CouponCodeRequestEventSerializer
+
+logger = logging.getLogger(__name__)
+
+
+# def generate_coupon_code_request_data(coupon_code_request):
+#     data = CouponCodeRequestData(
+#         uuid=str(coupon_code_request.uuid),
+#         lms_user_id=str(coupon_code_request.user.lms_user_id),
+#         course_id=str(coupon_code_request.course_id),
+#         enterprise_customer_uuid=str(coupon_code_request.enterprise_customer_uuid),
+#         state=coupon_code_request.state,
+#         reviewed_at=str(coupon_code_request.reviewed_at),
+#         reviewer_lms_user_id=str(coupon_code_request.reviewer.lms_user_id),
+#         coupon_id=str(coupon_code_request.coupon_id)
+#     )
+#     bridge = AvroAttrsBridge(CouponCodeRequestData)
+#     return bridge.serialize(data)
+
+# def generate_license_request_data(license_request):
+#     pass
+
+class ProducerFactory:
+    """
+    Factory class to create event producers.
+    The factory pattern is used to ensure only one producer per event type, which is the confluent recommendation
+    """
+    _type_to_producer = {}
+
+    @classmethod
+    def get_or_create_event_producer(cls, event_type, event_key_serializer, event_value_serializer):
+        """
+        Factory method to return the correct producer for the event type, or
+        create a new producer if none exists
+        :param event_type: name of event (same as segment events)
+        :param event_key_serializer:  AvroSerializer instance for serializing event key
+        :param event_value_serializer: AvroSerializer instance for serializing event value
+        :return: SerializingProducer
+        """
+        existing_producer = cls._type_to_producer.get(event_type)
+        if existing_producer is not None:
+            return existing_producer
+
+        producer_settings = {
+            'bootstrap.servers': getattr(settings, 'KAFKA_BOOTSTRAP_SERVER', ''),
+            'key.serializer': event_key_serializer,
+            'value.serializer': event_value_serializer,
+        }
+
+        if settings.KAFKA_API_KEY and settings.KAFKA_API_SECRET:
+            producer_settings.update({
+                'sasl.mechanism': 'PLAIN',
+                'security.protocol': 'SASL_SSL',
+                'sasl.username': getattr(settings, 'KAFKA_API_KEY', ''),
+                'sasl.password': getattr(settings, 'KAFKA_API_SECRET', ''),
+            })
+
+        new_producer = SerializingProducer(producer_settings)
+        cls._type_to_producer[event_type] = new_producer
+        return new_producer
+
+
+def create_topic_if_not_exists(topic_name):
```

Comment:
now that we have two topics it'd be more efficient to update this to take a list of topics instead of a single one since create_topics expects a list anyway